desktop/src/main/fileProcessing/WorkerPool.ts

```diff
@@ -25,166 +25,149 @@ type Task<T = unknown, U = unknown> = {
  * - Provides a method to shut down all active workers gracefully.
  */
 export class WorkerPool extends EventEmitter {
-  private maxWorkers: number;
-  private tasks: Task[];
-  private workers: Set<Worker>;
-  private workerMap: Map<string, Worker[]>; // For reusing workers
-
-  constructor(maxWorkers: number) {
-    super();
-    this.maxWorkers = maxWorkers;
-    this.tasks = [];
-    this.workers = new Set();
-    this.workerMap = new Map();
-  }
-
-  /**
-   * Adds a task to the queue and runs it when a worker is available
-   */
-  runTask<T, U>(
-    workerScript: string,
-    workerData: WorkerData<T>,
-    timeout?: number
-  ): Promise<U> {
-    return new Promise<U>((resolve, reject) => {
-      let timer: NodeJS.Timeout | undefined;
-
-      if (timeout !== undefined) {
-        timer = setTimeout(() => {
-          reject(new Error("Task timeout"));
-        }, timeout);
-      }
-
-      const worker = this.getOrCreateWorker(workerScript, workerData);
-      this.workers.add(worker);
-
-      worker.on("message", (message) => {
-        const task = workerScript.includes("metadata") ? "metadata" : "copy";
-        // Handle progress updates
-        if (message.type === "progress") {
-          this.emit("progress", {
-            task,
-            ...message,
-          });
-        }
-        // Handle invalid path updates
-        else if (message.type === "missingPath") {
```

Comment:
This change needs to be reverted. For example 'missingPath' was removed during merge